src/Compilers/Core/Portable/Text/StringText.cs

```diff
@@ -92,11 +92,17 @@ public override void Write(TextWriter textWriter, TextSpan span, CancellationTok
         {
             if (span.Start == 0 && span.End == this.Length)
             {
-                textWriter.Write(this.Source);
+                // Even on .NET Core, prefer the string overload, because in case the writer is a custom TextWriter that doesn't
+                // override the ReadOnlySpan<char> overload, it delegates to the array one under the hood.
+                textWriter.Write(_source);
```

Comment:
```suggestion
                textWriter.Write(this.Source);
```

Reply:
Why?

Reply:
because the change seems entirely stylistic with no impact on behavior (and as such isn't necessary for this change around improving performance).

Reply:
Done.